indico_toolkit/staggered_loop/staggered_loop.py

```diff
@@ -20,11 +43,484 @@ def __init__(self, client: IndicoClient):
+        for another environment (dev).
+
+        Args:
+            workflow_results: Workflow results to set as attribute of instance of class
+        """
+        self._workflow_results = workflow_results
+
+    def get_dataset_details(
+        self, dataset_id: int
+    ) -> Dict[str, Union[int, str, Dict[str, Union[int, str]]]]:
+        """
+        Find the following for the dataset in dev that we want to add data to
+        * datacolumns/labelsets
+        * updatedAt timestamp
+
+        Args:
+            dataset_id: Dataset ID to fetch details for
+
+        Returns:
+            dataset_details: JSON response from graphQL with dataset details
+
+        Raises:
+            ValueError: If no dataset details are returned for the supplied dataset_id
+        """
+        # Custom graphQL query to return updatedAt timestamp in addition to
+        # other details about the dataset
+        query = """
+        query getDataset($id: Int) {
+            dataset(id: $id) {
+                id
+                name
+                rowCount
+                status
+                updatedAt
+                datacolumns {
+                    id
+                    name
+                }
+                labelsets{
+                    id
+                    name
+                }
+            }
+        }
+        """
+        response = self.graphQL_request(
+            graphql_query=query, variables={"id": dataset_id}
+        )
+        dataset_details = response.get("dataset")
+        if not dataset_details:
+            raise ValueError(f"No dataset found for id={dataset_id}")
+        print(f"Fetched dataset metadata dataset_id={dataset_id}")
+
+        return dataset_details
+
+    def get_review_data(
+        self, workflow_id: int, update_date: datetime
+    ) -> List[WorkflowResult]:
+        """
+        Fetch review data from workflow, using the update_date as a filter
+
+        TODO Could be preferable to filter data by completedAt timestamp instead
+         of updatedAt, but that is more complex
+
+        Will also run pre review and post review comparisons here and
+        add TP/FP/FN metadata as well
+
+        Args:
+            workflow_id: ID of production workflow with review enabled
+            update_date: Update date for dataset in dev. Used to only capture
+                submissions that have been updated since then
+
+        Returns:
+            workflow_results: Review data fetched from workflow
+        """
+        # Get list of submission IDs using GetDocumentReport query
+        submission_ids = []
+        document_filter = DocumentReportFilter(
+            # TODO Comment back in when latest version of the client has been deployed
+            #  Earlier version of client does not have status as an argument to filter
+            # status="COMPLETED",
+            workflow_id=workflow_id,
+            updated_at_start_date=update_date,
```

Comment:
This `update_date` is so that we don't add review data that has already been added to the dev dataset. The update date is populated by looking at the last update timestamp on the dev dataset.

This could break down if the dataset was updated for another reason, in which case some review data that hasn't already been added to the dataset would be skipped by this filter.